In [7]:
import pandas as pd
import numpy as np


In [8]:
#load the contents of a file into a pandas Dataframe

fb = pd.read_csv('data/facebook.csv')
ms = pd.read_csv('data/microsoft.csv')

In [9]:
print(ms.iloc[0, 0])

2014-12-31


In [10]:
print(type(fb))

<class 'pandas.core.frame.DataFrame'>


In [11]:
fb.head()

Date       Open       High        Low      Close  Adj Close   Volume
0  2014-12-31  20.400000  20.510000  19.990000  20.049999  19.459270  4157500
1  2015-01-02  20.129999  20.280001  19.809999  20.129999  19.536913  2842000
2  2015-01-05  20.129999  20.190001  19.700001  19.790001  19.206934  4948800
3  2015-01-06  19.820000  19.840000  19.170000  19.190001  18.624611  4944100
4  2015-01-07  19.330000  19.500000  19.080000  19.139999  18.576082  8045200

In [12]:
 #Convert the 'Date' column to datetime format
fb['Date'] = pd.to_datetime(fb['Date'])

# Set the 'Date' column as the index
fb.set_index('Date', inplace=True)

The index of the data was random assigned primary keys. We have changed the index to be the date of the stock prices. To do so, first I had to convert the Date column to datetime format. 
- I then set the date column as the index. 

In [13]:
#getting the index of the fb dataframe 
print(fb.index)

DatetimeIndex(['2014-12-31', '2015-01-02', '2015-01-05', '2015-01-06',
               '2015-01-07', '2015-01-08', '2015-01-09', '2015-01-12',
               '2015-01-13', '2015-01-14',
               ...
               '2018-01-23', '2018-01-24', '2018-01-25', '2018-01-26',
               '2018-01-29', '2018-01-30', '2018-01-31', '2018-02-01',
               '2018-02-02', '2018-02-05'],
              dtype='datetime64[ns]', name='Date', length=780, freq=None)


In [14]:
#getting the columns of the dataframe
fb.columns

Index(['Open', 'High', 'Low', 'Close', 'Adj Close', 'Volume'], dtype='object')

In [15]:
fb.head()

Open       High        Low      Close  Adj Close   Volume
Date                                                                      
2014-12-31  20.400000  20.510000  19.990000  20.049999  19.459270  4157500
2015-01-02  20.129999  20.280001  19.809999  20.129999  19.536913  2842000
2015-01-05  20.129999  20.190001  19.700001  19.790001  19.206934  4948800
2015-01-06  19.820000  19.840000  19.170000  19.190001  18.624611  4944100
2015-01-07  19.330000  19.500000  19.080000  19.139999  18.576082  8045200

In [16]:
#getting the number of rows and columns
fb.shape

(780, 6)

In [17]:
#summary statistics of fb dataframe
fb.describe()

Open        High         Low       Close   Adj Close  \
count  780.000000  780.000000  780.000000  780.000000  780.000000   
mean    80.212705   81.285654   79.022397   80.264897   79.914215   
std     64.226121   65.048907   63.190963   64.198375   64.327846   
min     19.250000   19.500000   18.940001   19.139999   18.576082   
25%     25.525000   26.085000   24.845000   25.475000   25.134512   
50%     53.379999   54.034999   52.930000   53.420000   53.035403   
75%    113.322502  115.779999  110.297499  113.702501  113.261238   
max    245.770004  249.270004  244.449997  246.850006  246.850006   

             Volume  
count  7.800000e+02  
mean   1.204453e+07  
std    8.221848e+06  
min    1.311200e+06  
25%    7.215200e+06  
50%    9.728700e+06  
75%    1.408885e+07  
max    9.232320e+07

The above function shows that there are 780 entries in the dataset. The mean for open, high, low, close and adj close are tabulated above. 

In [18]:
#slicing dataframe using .loc method

fb.loc['2015-01-02', 'Close']

20.129999

In [19]:
#slicing using iloc method
# the 1 is the row and 3 is the column
# position starts with 0
fb.iloc[1, 3]

20.129999

In [20]:
#slicing a whole year
fb.loc['2015-01-01':'2015-12-31', 'Close']

Date
2015-01-02    20.129999
2015-01-05    19.790001
2015-01-06    19.190001
2015-01-07    19.139999
2015-01-08    19.860001
                ...    
2015-12-24    33.169998
2015-12-28    33.139999
2015-12-29    33.680000
2015-12-30    33.389999
2015-12-31    32.959999
Name: Close, Length: 252, dtype: float64

In [21]:
#selecting multiple columns 
fb.iloc[624: , : ]

Open        High         Low       Close   Adj Close  \
Date                                                                     
2017-06-23  158.679993  159.320007  153.220001  153.830002  153.587982   
2017-06-26  155.160004  156.600006  148.330002  152.149994  151.910614   
2017-06-27  151.440002  151.789993  146.350006  146.580002  146.349380   
2017-06-28  149.320007  151.940002  145.750000  151.750000  151.511246   
2017-06-29  150.600006  150.720001  144.080002  146.679993  146.449219   
...                ...         ...         ...         ...         ...   
2018-01-30  241.110001  246.419998  238.410004  242.720001  242.720001   
2018-01-31  245.770004  249.270004  244.449997  245.800003  245.800003   
2018-02-01  238.520004  246.899994  238.059998  240.500000  240.500000   
2018-02-02  237.000000  237.970001  231.169998  233.520004  233.520004   
2018-02-05  227.000000  233.229996  205.000000  213.699997  213.699997   

              Volume  
Date                  
2017-06-23  27214700  
2017-06-26  26599000  
2017-06-27  24987300  
2017-06-28  24873700  
2017-06-29  26610600  
...              ...  
2018-01-30  14270800  
2018-01-31  11964400  
2018-02-01  12980600  
2018-02-02  17961600  
2018-02-05  28869000  

[156 rows x 6 columns]

The above shows the columns from 624 to the end. The results are 156 rows. 

## Visualizing stock price